### DISCUSSION TITLE
Dollar sign get removed from process.argv on transpilation

### DISCUSSION THREAD

**Discussion by OlivierCuyp** [TypeStrong/ts-node#1875] [General]
### Search Terms

process.argv script argument dollar sign $

### Expected Behavior

Keep `$` in process.argv when it correctly passed to script argument.

### Actual Behavior

When creating a script that takes arguments, and if an argument is passed with a `$`, it disappear from it in process.argv.

### Steps to reproduce the problem

1. Create a script that logs process.argv
2. Run the script with an argument containing a `$`

### Minimal reproduction

dollar.ts:
```js
console.log(process.argv);
```

execution (in bash terminal):
```shell
> npx ts-node ./scripts/dollar.ts --test='$7test'
[
  '/Users/me/project/scripts/node_modules/.bin/ts-node',
  '/Users/me/project/scripts/dollar.ts',
  '--test=test' // <---  '$7' disappeared
]
```

In pure JS, there is no issue with the same script:

```shell
> node ./scripts/dollar.js --test='$7test'
[
  '/Users/me/project/scripts/node_modules/.bin/ts-node',
  '/Users/me/project/scripts/dollar.ts',
  '--test=$7test' // <--- all good !
]
```

### Specifications

* ts-node version:  10.9.1
* node version: 16.15.1
* TypeScript version: 4.8.2
* tsconfig.json, if you're using one:
```
{
  "ts-node": {
    "files": true,
    "transpileOnly": true
  },
  "compilerOptions": {
    "target": "es6",
    "module": "commonjs",
    "sourceMap": true,
    "strict": true,
    "esModuleInterop": true,
    "forceConsistentCasingInFileNames": true
  }
}
```
* package.json:
```
{
  "name": "dollar",
  "version": "1.0.0",
  "description": "",
  "main": "dollar.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "ts-node": "^10.9.1",
    "typescript": "^4.8.2"
  }
}
```
* Operating system and version: OSX 12.5.1

**Comment by cspotcode** [comment]
Converted to discussion because this is probably not a ts-node issue.  We've received a ton of issues that are caused by all the moving parts outside of ts-node's control, for example: npm, npx, the shim script that is created by npm / npx, your shell, etc.  If you're using 3 things at once, then you don't really know which one is causing the problem.

**Reply by OlivierCuyp** [reply]
@cspotcode Thanks I didn't think about npx side effect.
I just tried and you are right, it is him causing the issue:

```sh
node_modules/.bin/ts-node dollar.ts --test='$123'
[
  '/Users/olivier/Projects/Sagacify/dollar/node_modules/.bin/ts-node',
  '/Users/olivier/Projects/Sagacify/dollar/dollar.ts',
  '--test=$123'. // $ still there
]
```

But with npx:

```sh
❯ npx ts-node dollar.ts --test='$123'
[
  '/Users/olivier/Projects/Sagacify/dollar/node_modules/.bin/ts-node',
  '/Users/olivier/Projects/Sagacify/dollar/dollar.ts',
  '--test=23' // $1 disappeared
]
```
